serve_diffusion_component.py

```diff
@@ -1,24 +1,50 @@
-# ! pip install torch diffusers git+https://github.com/Lightning-AI/lightning-diffusion-component.git
-import diffusers
+# !pip install 'git+https://github.com/Lightning-AI/LAI-API-Access-UI-Component.git@diffusion'
+# !pip install 'git+https://github.com/Lightning-AI/stablediffusion.git@lit'
+# !curl https://raw.githubusercontent.com/runwayml/stable-diffusion/main/configs/stable-diffusion/v1-inference.yaml -o v1-inference.yaml
+
+
 import lightning as L
-import torch
+import base64, io, os, torch, lightning_diffusion
+from ldm.lightning import LightningStableDiffusion, PromptDataset
 
-from lightning_diffusion import BaseDiffusion, download_from_lightning_cloud
 
+class ServeDiffusion(lightning_diffusion.BaseDiffusion):
+    def __init__(self, input_type=L.app.components.Text, output_type=L.app.components.Image, **kwargs):
+        super().__init__(input_type=input_type, output_type=output_type, **kwargs)
+        self._model = None
 
-class ServeDiffusion(BaseDiffusion):
-    def setup(self, *args, **kwargs):
-        download_from_lightning_cloud(
-            "daniela/stable_diffusion", version="latest", output_dir="model"
+    def setup(self):
+        os.system(
+            "curl -C - https://pl-public-data.s3.amazonaws.com/dream_stable_diffusion/v1-5-pruned-emaonly.ckpt -o v1-5-pruned-emaonly.ckpt"
         )
-        self.model = diffusers.StableDiffusionPipeline.from_pretrained("model").to(
-            self.device
+        self._trainer = L.Trainer(
+            accelerator="auto",
+            devices=1,
+            precision=16 if torch.cuda.is_available() else 32,
+            enable_progress_bar=False,
+            inference_mode=False,
```

Comment:
you can change  `inference_mode=torch.cuda.is_available()`, to account for removing ` with torch.no_grad()`